src/Microsoft.AspNetCore.Authentication/ClaimsTransformationMiddleware.cs

```diff
@@ -35,17 +36,22 @@ public class ClaimsTransformationMiddleware
 
         public async Task Invoke(HttpContext context)
         {
-            var handler = new ClaimsTransformationHandler(Options.Transformer, context);
+            var transform = Options.Transformer;
+            if (transform == null && Options.TransformerType != null)
```

Comment:
One may want to use a transformer for a Web API sub-app (via `app.Map` or `app.UseWhen`) and another one for the rest of the app.